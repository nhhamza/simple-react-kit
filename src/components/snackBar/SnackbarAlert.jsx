import React from 'react';
import PropTypes from 'prop-types';
import Snackbar from '@material-ui/core/Snackbar';
import MuiAlert from '@material-ui/lab/Alert';
import { makeStyles } from '@material-ui/core/styles';
import styles from './style.jsx';

const SnackbarAlert = ({
  handleClose,
  open,
  message,
  severity,
  position = { vertical: 'bottom', horizontal: 'left' },
}) => {
  const classes = makeStyles(styles)();

  return (
    <div className={classes.root}>
      <Snackbar
        anchorOrigin={{
          vertical: position.vertical,
          horizontal: position.horizontal,
        }}
        open={open}
        autoHideDuration={2000}
        onClose={handleClose}
      >
        <MuiAlert
          elevation={6}
          variant="filled"
          onClose={handleClose}
          severity={severity}
        >
          <span className={classes.text}>{message}</span>
        </MuiAlert>
      </Snackbar>
    </div>
  );
};

SnackbarAlert.propTypes = {
  handleClose: PropTypes.func,
  open: PropTypes.bool,
  message: PropTypes.string,
  severity: PropTypes.string,
  position: PropTypes.objectOf(PropTypes.object()),
};

export default SnackbarAlert;
