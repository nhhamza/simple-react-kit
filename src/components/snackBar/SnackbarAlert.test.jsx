import React from 'react';

import { shallow } from 'enzyme';

import MuiAlert from '@material-ui/lab/Alert';
import SnackbarAlert from './SnackbarAlert.jsx';

describe('SnackbarAlert component', () => {
  const wrapper = shallow(<SnackbarAlert />);
  it('should render', () => {
    expect(wrapper.exists()).toMatchSnapshot();
  });

  it('should render span', () => {
    const EXPECTED_MESSAGE = 'hola';
    const MOCK_PROPS = {
      message: 'hola',
      open: true,
    };
    const newWrapper = shallow(<SnackbarAlert {...MOCK_PROPS} />);
    expect(
      newWrapper
        .find(MuiAlert)
        .dive()
        .find('span')
        .text(),
    ).toEqual(EXPECTED_MESSAGE);
  });
});
