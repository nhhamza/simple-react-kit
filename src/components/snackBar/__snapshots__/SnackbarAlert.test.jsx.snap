// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnackbarAlert component should render 1`] = `true`;
